packages/safe-tools-client/tests/acceptance/wallet-connection-test.ts

```diff
@@ -0,0 +1,54 @@
+import { truncateMiddle } from '@cardstack/ember-shared/helpers/truncate-middle';
+import { click, visit } from '@ember/test-helpers';
+
+import { module, test } from 'qunit';
+
+import {
+  FAKE_META_MASK_ACCOUNT,
+  FAKE_WALLET_CONNECT_ACCOUNT,
+  setupApplicationTest,
+} from '../helpers';
+
+module('Acceptance | wallet connection', function (hooks) {
+  setupApplicationTest(hooks);
+
+  module('With Metamask', function () {
+    test('connecting wallet', async function (assert) {
+      await visit('/schedule');
+      await click('.connect-button__button');
+      await click('[data-test-wallet-option="metamask"]');
+
+      assert.dom(
+        '.boxel-radio-option__input boxel-radio-option__input--hidden-radio boxel-radio-option__input--checked'
+      );
+      await click('[data-test-mainnet-connect-button]');
+
+      assert
+        .dom('.safe-tools__dashboard-schedule-control-panel-wallet-address')
+        .hasText(truncateMiddle([FAKE_META_MASK_ACCOUNT]));
+    });
+  });
+
+  module('With Wallet Connect', function () {
+    test('connecting wallet', async function (assert) {
+      await visit('/schedule');
+      await click('.connect-button__button');
+
+      await click('[data-test-wallet-option="wallet-connect"]');
+
+      assert.dom(
+        '.boxel-radio-option__input boxel-radio-option__input--hidden-radio boxel-radio-option__input--checked'
+      );
+
+      await click('[data-test-mainnet-connect-button]');
+
+      this.mockWalletConnect.mockConnectedWallet([FAKE_WALLET_CONNECT_ACCOUNT]);
+      this.mockWalletConnect.mockAccountsChanged([FAKE_WALLET_CONNECT_ACCOUNT]);
+
+      await click('.network-connect-modal__close-button'); // FIXME: I don't think this click should be necessary
```

Comment:
Agreed, I just didn’t want to get too deep into adding functionality like this until we could have tests around it, so thanks!